lib/superagent-mock.js

```diff
@@ -47,6 +47,45 @@ function mock (superagent, config, logger) {
   }();
 
   /**
+   * Chains progrss events and eventually returns the response
+   * @param ProgressEvent (object/function) the ProgressEvent implementation
+   * @param request (object) the current request
+   *
+   * @param progress (object) the progress configuration
+   * @param progress.parts (number)
+   * @param progress.delay (number)
+   * @param progress.total (number)
+   * @param progress.lengthComputable (boolean)
+   * @param progress.direction (string)
+   *
+   * @param remainingParts (number) the remaining calls to make
+   * @param callback (function) the end response callback
+   */
+  var handleProgress = function (ProgressEvent, request, progress, remainingParts, callback) {
+    setTimeout(function () {
+      var total = progress.total || 100;
+      var loaded = total * (progress.parts - remainingParts) / progress.parts;
+      var lengthComputable = typeof progress.lengthComputable !== 'undefined' ? !!progress.lengthComputable : true;
```

Comment:
I would say you have a lot of checks on the context.progress configuration instead of applying a default config, here and below (line 184). Would you like refactoring it ? I can do it as well.